docs/Bitbucket.md

```diff
@@ -0,0 +1,107 @@
+| `pipelines.default.step.clone.sparse-checkout.patterns` | ✅ | The list of paths to invoke a sparse checkout for. Can be a pattern to a specific file, directory, or wildcard for all files belonging within a certain directory. |
+| `pipelines.default.step.deployment` | ❌ | The environment set for the Bitbucket Deployments dashboard. This has no translatable equivalent within Buildkite. |
+| `pipelines.default.step.docker` | ❌ | The availability of docker in a specific Bitbucket pipeline step. This will depend on the agent configuration that the corresponding Buildkite command step is being targeted to run said job has available. Consider [tagging](https://buildkite.com/docs/agent/v3/cli-start#tags) agents with `docker=true` to ensure Buildkite command steps requiring hosts with Docker installed and configured to accept/run specific jobs. |
+| `pipelines.default.step.fail-fast` | ❌ | Whether a specific step of a Bitbucket pipeline allows a parallel step to fail entirely if it fails (set as `true`), or allows failures (set as `false`). Consider using a combination of `soft_fail` and/or `cancel_on_build_failing` in the corresponding Buildkite command steps' [attributes](https://buildkite.com/docs/pipelines/command-step#command-step-attributes) for a similar [approach](https://buildkite.com/docs/pipelines/command-step#fail-fast). |
+| `pipelines.default.step.image` | ✅ | The container image that is to be applied for a specific step within a Bitbucket pipeline. Images set at this level will be applied irrespective of the pipeline-level `image` key that is set, and will be applied in the corresponding Buildkite pipeline using the [docker-buildkite-plugin](https://github.com/buildkite-plugins/docker-buildkite-plugin). The `name`, `username` and `password` properties below are the alternate way of defining a Bitbucket pipeline step to use a specific image (with authentication). |
+| `pipelines.default.step.image.name` | ✅ | The name of the image to utilise within the Bitbucket pipeline step. Translated to the `image` property within the [docker-buildkite-plugin](https://github.com/buildkite-plugins/docker-buildkite-plugin) (if the image name was already not set in the `pipelines.default.step.image` property above itself.) |
+| `pipelines.default.step.image.username` | ✅ | The username of the user for private Docker images hosted on a given registry. Translated to the `username` property within the [docker-login-buildkite-plugin](https://github.com/buildkite-plugins/docker-login-buildkite-plugin) |
+| `pipelines.default.step.image.password` | ✅ | The password of the user for private Docker images hosted on a given registry. Translated to the `password` property within the [docker-login-buildkite-plugin](https://github.com/buildkite-plugins/docker-login-buildkite-plugin). Be sure to utilise [best practices](https://buildkite.com/docs/pipelines/secrets) for storing and accessing secrets within a Buildkite pipeline! |
```

Comment:
I would just include the main `image` key here and link to the general option for more information

Reply:
Tweaked this part (and also the parent `image` key) entirely with the sub-property support instead of seperate rows